src/app/components/elements/panels/SelectedEventDetails.tsx

```diff
@@ -30,6 +30,32 @@ export const countStudentsAndTeachers = (eventBookings: EventBookingDTO[]) => {
   };
 };
 
+export const countGenders = (eventBookings: EventBookingDTO[]) => {
+  const genders = {
+    male: 0,
+    female: 0,
+    other: 0,
+    preferNotToSay: 0,
+    unknown: 0,
+  };
+
+  eventBookings.forEach((booking) => {
+    const gender = booking.userBooked?.gender;
+    const bookingStatus = booking.bookingStatus;
+
+    if (gender && bookingStatus) {
```

Comment:
Question: if a gender is `undefined` should that count as `UNKNOWN`?

Reply:
In theory it shouldn't be undefined (unless something was wrong on the API side and didn't send it!) because we changed it so that gender is required at registration, and legacy users without a gender get an annoying popup until they select one. However I've added it into the countGenders function and added a test.

Similarly *technically* we shouldn't ever see `UNKNOWN` gender being sent from the API for Event Bookings because UNKNOWN is used for anonymous users who can't book onto events, but it's covered anyway just in case.

Reply:
Thanks for clarifying. If you know that the API will never send an undefined gender, doesn't that mean that `UserSummaryWithEmailAddressAndGenderDTO` should have `gender` as non optional? Like this:

```typescript
gender: Gender;
```

Reply:
It *shouldn't* be sending undefined gender simply because we made the popup super annoying for legacy accounts, but it is technically possibly for them to continually ignore the account update popup and continue using the site without gender selected (in which case gender would be undefined). I suspect the majority of users would get annoyed by the popup and just update the gender, even if they just selected "prefer not to say", but I can't guarantee that, hence it's optional.

Reply:
Perfect, that makes sense, thank you!